crates/node-builder/src/builder.rs

```diff
@@ -469,334 +451,59 @@ where
-            client,
-            pipeline,
-            blockchain_db.clone(),
-            Box::new(executor.clone()),
-            Box::new(network.clone()),
-            max_block,
-            config.debug.continuous,
-            payload_builder.clone(),
-            initial_target,
-            reth_beacon_consensus::MIN_BLOCKS_FOR_PIPELINE_RUN,
-            consensus_engine_tx,
-            consensus_engine_rx,
-            hooks,
-        )?;
-        info!(target: "reth::cli", "Consensus engine initialized");
-
-        let events = stream_select!(
-            network.event_listener().map(Into::into),
-            beacon_engine_handle.event_listener().map(Into::into),
-            pipeline_events.map(Into::into),
-            if config.debug.tip.is_none() && !config.dev.dev {
-                Either::Left(
-                    ConsensusLayerHealthEvents::new(Box::new(blockchain_db.clone()))
-                        .map(Into::into),
-                )
-            } else {
-                Either::Right(stream::empty())
-            },
-            pruner_events.map(Into::into),
-            static_file_producer_events.map(Into::into)
-        );
-        executor.spawn_critical(
-            "events task",
-            reth_node_core::events::node::handle_events(
-                Some(network.clone()),
-                Some(head.number),
-                events,
-                database.clone(),
-            ),
-        );
-
-        let engine_api = EngineApi::new(
-            blockchain_db.clone(),
-            config.chain.clone(),
-            beacon_engine_handle,
-            payload_builder.into(),
-            Box::new(executor.clone()),
-        );
-        info!(target: "reth::cli", "Engine API handler initialized");
-
-        // extract the jwt secret from the args if possible
-        let default_jwt_path = data_dir.jwt_path();
-        let jwt_secret = config.rpc.auth_jwt_secret(default_jwt_path)?;
-
-        // adjust rpc port numbers based on instance number
-        config.adjust_instance_ports();
-
-        // Start RPC servers
-
-        let (rpc_server_handles, mut rpc_registry) = crate::rpc::launch_rpc_servers(
-            node_components.clone(),
-            engine_api,
-            &config,
-            jwt_secret,
-            rpc,
-        )
-        .await?;
+    >
+    where
+        Types: Node<
+            FullNodeTypesAdapter<Types, DB, RethFullProviderType<DB, <Types as NodeTypes>::Evm>>,
+        >,
+        Types::PoolBuilder: PoolBuilder<RethFullAdapter<DB, Types>>,
+        Types::NetworkBuilder: crate::components::NetworkBuilder<
+            RethFullAdapter<DB, Types>,
+            <Types::PoolBuilder as PoolBuilder<RethFullAdapter<DB, Types>>>::Pool,
+        >,
+        Types::PayloadBuilder: crate::components::PayloadServiceBuilder<
+            RethFullAdapter<DB, Types>,
+            <Types::PoolBuilder as PoolBuilder<RethFullAdapter<DB, Types>>>::Pool,
+        >,
```

Comment:
hmm, why is is this necessary now?

Reply:
I'll take another look at this

Reply:
cool, I'm still very supportive of this feature just haven't entirely settled on a design yet, but I think the Launch trait is the best we can do atm